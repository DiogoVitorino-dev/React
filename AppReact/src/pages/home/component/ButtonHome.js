import React from 'react';
import {Button} from 'react-native';

export default function ButtonHome({disabled, onPressButton}) {
  return (
    <Button
      title="add fifty"
      onPress={() => onPressButton(50)}
      color={disabled ? 'tomato' : 'turquoise'}
    />
  );
}
