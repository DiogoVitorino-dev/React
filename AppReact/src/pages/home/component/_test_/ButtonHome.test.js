import { describe, expect, it, jest, test } from "@jest/globals";
import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ButtonHome from "../ButtonHome";

describe('ButtonHome Component',() => {
    it('renders correctly', () => {
        const tree = renderer.create(<ButtonHome />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should call onPress prop on click', () => {
        const fn = jest.fn()
        render(<ButtonHome disabled={false} onPressButton={fn}>Click</ButtonHome>)
        const button = screen.getByRole('button')
        
        fireEvent.press(button)

        expect(fn).toBeCalledWith(50)
    })
})
