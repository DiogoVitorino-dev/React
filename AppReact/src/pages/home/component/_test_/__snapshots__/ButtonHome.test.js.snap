// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonHome Component renders correctly 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
          Object {
            "color": "turquoise",
          },
        ]
      }
    >
      add fifty
    </Text>
  </View>
</View>
`;
